src/model.jl

```diff
@@ -609,14 +676,13 @@ function _build_model(
     # Positional indices from mpc.gen
     PMAX = 9
     PMIN = 10
-
     println("building sets: ", Dates.now())
     # Sets - time periods
     hour_idx = 1:interval_length
     end_index = start_index + interval_length - 1
     # Sets - static
-    sets = _make_sets(case, storage)
-
+    print(123)
+    sets = _make_sets(case, storage, demand_flexibility)
```

Comment:
You mentioned on the call that there was some problem you encountered with the `_make_sets` function. Could you please tell us a little more about what you were trying to do, and what error you encountered?

Reply:
> You mentioned on the call that there was some problem you encountered with the `_make_sets` function. Could you please tell us a little more about what you were trying to do, and what error you encountered?

For example, I have experimented putting line 264-268: 
>          flexible_load_bus_map = sparse(
>                sets.flexible_bus_idx_up, 1:sets.num_flexible_bus_up, 1, sets.num_bus, sets.num_flexible_bus_up
>            )::SparseMatrixCSC
in to `_make_sets()` function and in type.jl add a new field `flexible_load_bus_map`. But the code gives a strange error when running in to the end of `_make_sets()` function where the sets is created `sets = Sets(; ... )`. :
![image](https://user-images.githubusercontent.com/77214628/128979600-6ca86168-e6a9-4926-94ce-969e1df90e38.png)

My workaround is putting the calculation inside fcn ` _add_constraint_power_balance!()` and everything works as expected.  This does not affect the flow and functionality of the code but I am just curious about what is causing it.

Reply:
This message is due to the `try`/`finally` logic within **gurobi.jl**. If an error is encountered within the l.14 call to `run_scenario`, it is caught and ignored by the `try` block, then within the `finally` block, the variable `m` has not been created and therefore can't be cleaned up. If you temporarily comment out the `try`/`finally`/`end` statements in l.13, l.16, and l.20, then you will be able to see the exception and can debug.